src/iop/denoiseprofile.c

```diff
@@ -1182,179 +1281,68 @@ static void process_wavelets(struct dt_iop_module_t *self, dt_dev_pixelpipe_iop_
-      // for instance, largest scale is DT_IOP_DENOISE_PROFILE_BANDS
-      // max_scale only indicates the number of scales to process at THIS
-      // zoom level, it does NOT corresponds to the the maximum number of scales.
-      // in other words, max_scale is the maximum number of VISIBLE scales.
-      // That is why we have this "scale+offset_scale"
-      float band_force_exp_2
-          = d->force[DT_DENOISE_PROFILE_ALL][DT_IOP_DENOISE_PROFILE_BANDS - (scale + offset_scale + 1)];
-      band_force_exp_2 *= band_force_exp_2;
-      band_force_exp_2 *= 4;
-      for(int ch = 0; ch < 3; ch++)
-      {
-        adjt[ch] *= band_force_exp_2;
-      }
-      band_force_exp_2 = d->force[DT_DENOISE_PROFILE_R][DT_IOP_DENOISE_PROFILE_BANDS - (scale + offset_scale + 1)];
-      band_force_exp_2 *= band_force_exp_2;
-      band_force_exp_2 *= 4;
-      adjt[0] *= band_force_exp_2;
-      band_force_exp_2 = d->force[DT_DENOISE_PROFILE_G][DT_IOP_DENOISE_PROFILE_BANDS - (scale + offset_scale + 1)];
-      band_force_exp_2 *= band_force_exp_2;
-      band_force_exp_2 *= 4;
-      adjt[1] *= band_force_exp_2;
-      band_force_exp_2 = d->force[DT_DENOISE_PROFILE_B][DT_IOP_DENOISE_PROFILE_BANDS - (scale + offset_scale + 1)];
-      band_force_exp_2 *= band_force_exp_2;
-      band_force_exp_2 *= 4;
-      adjt[2] *= band_force_exp_2;
-    }
-    else
-    {
-      float band_force_exp_2 = d->force[DT_DENOISE_PROFILE_Y0][DT_IOP_DENOISE_PROFILE_BANDS - (scale + offset_scale + 1)];
-      band_force_exp_2 *= band_force_exp_2;
-      band_force_exp_2 *= 4;
-      adjt[0] *= band_force_exp_2;
-      band_force_exp_2 = d->force[DT_DENOISE_PROFILE_U0V0][DT_IOP_DENOISE_PROFILE_BANDS - (scale + offset_scale + 1)];
-      band_force_exp_2 *= band_force_exp_2;
-      band_force_exp_2 *= 4;
-      adjt[1] *= band_force_exp_2;
-      adjt[2] *= band_force_exp_2;
-    }
-
-    const float DT_ALIGNED_PIXEL thrs[4] = { adjt[0] * sb2 / std_x[0],
-                                             adjt[1] * sb2 / std_x[1],
-                                             adjt[2] * sb2 / std_x[2],
-                                             0.0f };
-// const float std = (std_x[0] + std_x[1] + std_x[2])/3.0f;
-// const float thrs[4] = { adjt*sigma*sigma/std, adjt*sigma*sigma/std, adjt*sigma*sigma/std, 0.0f};
-// fprintf(stderr, "scale %d thrs %f %f %f = %f / %f %f %f \n", scale, thrs[0], thrs[1], thrs[2], sb2,
-// std_x[0], std_x[1], std_x[2]);
-#endif
     const float DT_ALIGNED_PIXEL boost[4] = { 1.0f, 1.0f, 1.0f, 1.0f };
-    synthesize(buf2, buf1, buf[scale], thrs, boost, width, height);
+    float DT_ALIGNED_PIXEL thrs[4];
+    variance_stabilizing_xform(thrs,scale,max_scale,npixels,sum_y2,d);
+    synthesize(out, out, buf, thrs, boost, width, height);
 
     float *buf3 = buf2;
     buf2 = buf1;
     buf1 = buf3;
   }
 
+  // add in the final residue
+#ifdef _OPENMP
+#pragma omp simd aligned(buf1, out : 64)
+#endif
+  for (size_t k = 0; k < 4 * npixels; k++)
```

Comment:
I think this one can bite :/ it's likely to cause:

(size_t) < (int) * (int)(size_t)

Reply:
The (int) in (int)*(size_t) gets promoted...